### PR TITLE
kubectl: add test coverage for autoscale command

### PR DESCRIPTION
**Pull request by omerap12**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Added test coverage for the `kubectl autoscale` command.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by omerap12** [comment]
cc @ardaguclu

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "custom-name",
+				},
+				Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
+					ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
+						APIVersion: "apps/v1",
+						Kind:       "Deployment",
+						Name:       "deployment-1",
+					},
+					MinReplicas:                    convertToInt32Pointer(2),
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:127]:
```suggestion
					MinReplicas:                    ptr.To(2),
```

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+					MaxReplicas:                    int32(10),
+					TargetCPUUtilizationPercentage: convertToInt32Pointer(80),
+				},
+			},
+		},
+		{
+			name: "create without max replicas",
+			options: &AutoscaleOptions{
+				Name:       "custom-name-3",
+				Max:        -1,
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:169]:
```suggestion
				Max:        nil,
```

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+type createHorizontalPodAutoscalerTestCase struct {
+	name        string
+	options     *AutoscaleOptions
+	refName     string
+	mapping     *meta.RESTMapping
+	expectedHPA *autoscalingv1.HorizontalPodAutoscaler
+}
+
+func convertToInt32Pointer(i int32) *int32 {
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:95]: Redundant, `ptr.To` already achieves this.

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+limitations under the License.
+*/
+
+package autoscale
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:23]: nit: New line after this to separate it from k8s dependency group.

**Review by ardaguclu — COMMENTED** [review]
Thank you
/triage accepted
/priority backlog

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go (hunk 4 above)
Comment (omerap12) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:23]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go (hunk 3 above)
Comment (omerap12) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:95]: 👍

File: staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go (hunk 2 above)
Comment (omerap12) [reply staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale_test.go:169]: Cannot use nil as int32 value:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go#L70

**Pull request merged**